adapters/repos/db/lsmkv/global_bucket_registry.go

```diff
@@ -0,0 +1,56 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package lsmkv
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+)
+
+type globalBucketRegistry struct {
+	buckets map[string]struct{}
+	mu      sync.Mutex
+}
```

Comment:
I think the registry could be removed but if it's left as a safe fuse, it may be associated to a store rather being a global instance. That would require some changes in the store down to the bucket.
We should improve some design aspects and we may include this in that discussion @parkerduckworth